From b0772404d3daba9c5edafd3b4f18e2deb1995c20 Mon Sep 17 00:00:00 2001
From: Tapasweni Pathak <tapaswenipathak@gmail.com>
Date: Thu, 7 Mar 2019 06:02:26 +0530
Subject: [PATCH] net: ipv6: Fix possible null dereference

xfrm_offload function can return null, there should be a null check
before dereferencing xo.
---
 net/ipv6/xfrm6_mode_tunnel.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/net/ipv6/xfrm6_mode_tunnel.c b/net/ipv6/xfrm6_mode_tunnel.c
index de1b0b8..aa834c5 100644
--- a/net/ipv6/xfrm6_mode_tunnel.c
+++ b/net/ipv6/xfrm6_mode_tunnel.c
@@ -113,6 +113,9 @@ static void xfrm6_mode_tunnel_xmit(struct xfrm_state *x, struct sk_buff *skb)
 {
 	struct xfrm_offload *xo = xfrm_offload(skb);
 
+	if (!xo)
+		return;
+
 	if (xo->flags & XFRM_GSO_SEGMENT)
 		skb->transport_header = skb->network_header + sizeof(struct ipv6hdr);
 
-- 
2.10.0

